cmd/minikube/cmd/start.go

```diff
@@ -177,7 +177,7 @@ func initKubernetesFlags() {
 	startCmd.Flags().Var(&node.ExtraOptions, "extra-config",
 		`A set of key=value pairs that describe configuration that may be passed to different components.
 		The key should be '.' separated, and the first part before the dot is the component to apply the configuration to.
-		Valid components are: kubelet, kubeadm, apiserver, controller-manager, etcd, proxy, scheduler
+		Valid components are: kubelet, kubeadm, apiserver, controller-manager, etcd, kube-proxy, scheduler
```

Comment:
optional: 
how about putting these valid options in a list, somwhere in the packge and call it here?
so it is always up to date ? and also when we validate the options, we can use the same

Reply:
@medyagh good idea, and been thinking of using something similar to kubeadm options for kubeproxy options also

```go
var KubeadmExtraArgsWhitelist = map[int][]string{
	KubeadmCmdParam: {
		"ignore-preflight-errors",
		"dry-run",
		"kubeconfig",
		"kubeconfig-dir",
		"node-name",
		"cri-socket",
		"experimental-upload-certs",
		"certificate-key",
		"rootfs",
		"skip-phases",
	},
	KubeadmConfigParam: {
		"pod-network-cidr",
	},
}
```

so it can be restricted in order to avoid failure when launching minikube, possible valid values for kubeproxy are as follows

```yaml
apiVersion: kubeproxy.config.k8s.io/v1alpha1
kind: KubeProxyConfiguration
bindAddress: ""
bindAddressHardFail: false
clientConnection:
    acceptContentTypes: ""
    burst: 0
    contentType: ""
    kubeconfig: ""
    qps: 0
clusterCIDR: ""
configSyncPeriod: 0s
conntrack:
    maxPerCore: null
    min: null
    tcpCloseWaitTimeout: null
    tcpEstablishedTimeout: null
detectLocalMode: ""
enableProfiling: false
healthzBindAddress: ""
hostnameOverride: ""
iptables:
    masqueradeAll: false
    masqueradeBit: null
    minSyncPeriod: 0s
    syncPeriod: 0s
ipvs:
    excludeCIDRs: null
    minSyncPeriod: 0s
    scheduler: ""
    strictARP: false
    syncPeriod: 0s
    tcpFinTimeout: 0s
    tcpTimeout: 0s
    udpTimeout: 0s
metricsBindAddress: ""
mode: ""
nodePortAddresses: null
oomScoreAdj: null
portRange: ""
showHiddenMetricsForVersion: ""
udpIdleTimeout: 0s
winkernel:
    enableDSR: false
    networkName: ""
    sourceVip: ""
```

but package only manages `key: value` for `KubeProxyConfiguration` since `extraOptions.AsMap().Get` only returns `map[string]string`, so as i mentioned a list with valid options for kube-proxy can be added

- bindAddress
- bindAddressHardFail
- configSyncPeriod
- detectLocalMode
- enableProfiling
- healthzBindAddress
- hostnameOverride
- mode
- nodePortAddresses
- oomScoreAdj
- portRange
- showHiddenMetricsForVersion
- udpIdleTimeout

Reply:
that sounds like a good idea for a recator. how about create an issue from your comment and make a follow up refactor PR so I see it in action what u mean ?

Reply:
@medyagh kk will do so!